css-variables/readme.md

```diff
@@ -0,0 +1,443 @@
+  --btn-bg-color: var(--primary-color);
+  --btn-fg-color: var(--primary-text-color);
+}
+
+.btn-negative {
+  --btn-bg-color: var(--negative-color);
+  --btn-fg-color: var(--negative-text-color);
+}
+```
+
+```jsx
+<button class="btn btn-primary"/>
+<button class="btn btn-negative"/>
+```
+
+
+## Inheritance
+
+CSS variables can be defined either in the root scope, or in a selector. We **use the root scope for global theme variables**, and in **the selector for component specific variables**. Inheritance works as expected, so if a component doesn't define a variable, it will inherit the value from the root scope. Use fallbacks pattern, in the case of something you are considering a semi-private API, and prefix with `--dh` to avoid conflicts (global).
+
+```scss
+// global theme
+:root {
+  --global-variable: red;
+}
+```
+
+```scss
+// MyComponent.scss
+.my-component {
+  --local-color: var(--global-variable);
+
+  .child {
+    color: var(--local-color);
+  }
+
+  .child.a {
+    --local-color: salmon;
+  }
+
+  .child.b {
+    // dh-child-b could inherit from the root scope, and thought of as a semi-private API
+    // in the sense that it's not published in the theme files, but it is in the global scope
+    --local-color: var(--dh-child-b, lemonchiffon);
+  }
+}
+```
+
+
+## Variables in JS
+
+A challenge with that is, monaco, ploly and our grid all require resolved javascript values for theming. We accomplish this by using JS to read the value of each css variable defined in our theme (for numbers, it getProperty, and colors by setting a background and reading the color back from the element).
+
+```js
+:root {
+  --color: red;
+}
+
+const propetry = getComputedStyle(document.documentElement).getPropertyValue('--color');
+// "red"
+const div = document.createElement('div');
+div.style.backgroundColor = 'var(--color)';
+const resolved_color = getComputedStyle(div).style.backgroundColor;
+// "rgb(255, 0, 0)" which we then normalize with JS to hex for use in monaco, plotly, and our grid.
+```
+
+Aside CSS properties are actually a great way to get values _from_ JS into CSS:
+
+[Codepen Example](https://codepen.io/dsmmcken/pen/ZEPjBmy)
+
+```js
+document.addEventListener("mousemove", (e) => {
+  document.documentElement.style.setProperty("--x", e.clientX);
+  document.documentElement.style.setProperty("--y", e.clientY);
+});
+```
+
+```scss
+div {
+  transform: translate(var(--x) * 1px, var(--y) * 1px);
+}
+```
+
+
+
+
+
+
+
+
+## Start using css variables!
+
+We have a theme component which manages which theme is applied. All users created themes must inheriet from either a light or a dark theme.
+
+1. All Deephaven **theme variables** start with `--dh` to avoid conflicts with other css variables. `--dh` variables should be thought of as a public API. Removing a `--dh` variable is a breaking change. Adding a variable is not a breaking change, as users themes inherit from the base themes, and will not break if a variable is added.
```

Comment:
```suggestion
1. All Deephaven **theme variables** start with `--dh` to avoid conflicts with other CSS variables. `--dh` variables should be thought of as a public API. Removing a `--dh` variable is a breaking change. Adding a variable is not a breaking change, as users themes inherit from the base themes, and will not break if a variable is added.
```